use vercel fork

dario-piotrowicz opened the pull request:
This PR makes Pages CI build using the new v2 beta image work.

The issue is that the current Vercel CLI opens too many file handlers at once (which the v2 beta image doesn't seem to be handling correctly) when building a Next.js >= v.13.3.0 app that contains prerendered routes.

The prerendered routes are problematic because as part of the building process Vercel generates lambda functions for them, such lambda functions include node modules, the parallel downloading of such node modules causes too many files to be open at once.

This PR introduces as a temporary fix the use of a fork of the Vercel CLI that sequentializes the above mentioned problematic downloads solving the issue (see https://github.com/dario-piotrowicz/vercel/commit/4f6355174fbe9717d8b85c0bec9329b7a0d3de60). 

This is meant to be a very temporary fix to be in place only until the v2 image has been fixed/improved to handle more files handlers at once. 

____

I've tested it with the following apps from my [testing apps repo](https://github.com/dario-piotrowicz/next-apps-for-testing):
- [middleware-13.2.4](https://github.com/dario-piotrowicz/next-apps-for-testing/tree/master/apps/middleware-13.2.4)
https://d8933adb.next-on-pages-ci-testing-app.pages.dev/
- [simple-app-dir-13.4-server-actions](https://github.com/dario-piotrowicz/next-apps-for-testing/tree/master/apps/simple-app-dir-13.4-server-actions)
https://2283cf03.next-on-pages-ci-testing-app.pages.dev/
- [simple-src-13.2.4](https://github.com/dario-piotrowicz/next-apps-for-testing/tree/master/apps/simple-src-13.2.4)
https://c97fecc1.next-on-pages-ci-testing-app.pages.dev/
- [simple-app-dir-13.3.1](https://github.com/dario-piotrowicz/next-apps-for-testing/tree/master/apps/simple-app-dir-13.3.1)
https://0a517f38.next-on-pages-ci-testing-app.pages.dev/
- [dynamic-static-app-dir-13.2.4](https://github.com/dario-piotrowicz/next-apps-for-testing/tree/master/apps/dynamic-static-app-dir-13.2.4)
https://7a1e4ab6.next-on-pages-ci-testing-app.pages.dev/

petebacondarwin commented:
IMO this PR (and maybe even the changeset) needs a bit more information about what this fork is, what it does, and why it is needed.

GregBrimble commented on README.md:
````diff
@@ -97,12 +97,12 @@ Then in a separate terminal run:
-### Install `@cloudflare/next-on-pages` and vercel (optional)
+### Install `@cloudflare/next-on-pages` and next-on-pages-vercel-cli (optional)
 
-To speed up local development (especially the refresh speed when running in watch mode) you can optionally choose to install `@cloudflare/next-on-pages` and `vercel` as dev dependencies of your project:
+To speed up local development (especially the refresh speed when running in watch mode) you can optionally choose to install `@cloudflare/next-on-pages` and `next-on-pages-vercel-cli` as dev dependencies of your project:
 
 ```sh
-npm install -D @cloudflare/next-on-pages vercel
+npm install -D @cloudflare/next-on-pages next-on-pages-vercel-cli
````
I thought this was just a temporary thing to get it working in Pages' v2 build system? I wouldn't document it — especially if it's just for local development and unnecessary there.

dario-piotrowicz replied:
> Where is the fork? If not in the Cloudflare org, I think it should be open-source at least somewhere (your personal account?) so inquisitive folks can see what code we're running.

It's in the PR's description (I had the link wrong, I've just updated it 🙇) I hope that's enough

dario-piotrowicz replied:
yeah it's totally temporary, so I updated this because I don't want to keep the vercel part while its going to be totally inaccurate (but I kept things rather vague 😅)

I'll fix this thanks 🙂

dario-piotrowicz replied:
I'll just (temporarily) remove this whole section is that ok by you 😕

GregBrimble commented on package.json:
```diff
@@ -33,7 +33,7 @@
 	},
 	"peerDependencies": {
-		"vercel": "^28.0.2",
+		"next-on-pages-vercel-cli": "latest",
```
Given this is a hacky thing and we're somewhat distrusting of it, I think pinning to a specific version would be good here. Let's us very precisely control the dependency on this fork, so we know exactly what version we're running of `@cloudflare/next-on-pages` and of the Vercel CLI fork.

GregBrimble commented on tests/_helpers/index.ts:
I'm guessing these changes came from bumping to vitest@0.31.0?

dario-piotrowicz replied:
yeah, without bumping the version it seemed like vitest was not correctly mocking esbuild and causing the tests to fail 😢

GregBrimble reviewed (changes requested):
I'd for sure pin the version, and I'd strongly consider removing any reference to this in public docs.

GregBrimble commented:
Where is the fork? If not in the Cloudflare org, I think it should be open-source at least somewhere (your personal account?) so inquisitive folks can see what code we're running.

dario-piotrowicz commented:
> Where is the fork? If not in the Cloudflare org, I think it should be open-source at least somewhere (your personal account?) so inquisitive folks can see what code we're running.

It's in the PR's description (I had the link wrong, I've just updated it 🙇) I hope that's enough

GregBrimble commented:
This `console.log` intentional? https://github.com/vercel/vercel/compare/main...dario-piotrowicz:vercel:next-on-pages-mod#diff-e38f85b3385148c0b7d002fe5ec2e5f8d8488e9a786f91cba00eeb556da437b4R552

dario-piotrowicz commented:
> This `console.log` intentional? [vercel/vercel@main...dario-piotrowicz:vercel:next-on-pages-mod#diff-e38f85b3385148c0b7d002fe5ec2e5f8d8488e9a786f91cba00eeb556da437b4R552](https://github.com/vercel/vercel/compare/main...dario-piotrowicz:vercel:next-on-pages-mod#diff-e38f85b3385148c0b7d002fe5ec2e5f8d8488e9a786f91cba00eeb556da437b4R552)

no sorry it's a leftover of my debugging 😓, I've removed that, updated the next-on-pages dependency and updated the PR's description with the latest commit, thanks for noticing it 🙂
